### PR TITLE
Feature/user tags filter.jw

### PR DESCRIPTION
**Pull request by jwasmer**
What does this PR do?
- Allows tags to be used on both the All Recipes tab and on the Cookbook tab.
- Updates recipes shown when tab is changed
- Resets tag selection when tab is changed
- Refactors tag functions for DRYness

Where should your reviewer start?
- tagsToggleFilter(), review the chain of functions that executes there
- Review event listener additions

What (if any) features are you implementing?
- Search by tags for users, tag reset when window selection changes

What (if anything) did you refactor?
- scripts.js

Were there any issues that arose?

Is there anything that you need from your teammate?

Any other comments, questions, or concerns to add?
- None of the existing functionality was changed, but some of it was refactored.

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -142,40 +140,98 @@ function buildTags(recipe) {
...
+    tagElement.innerText = tag;
+
+    tagElement.onclick = tagsToggleFilter;
+
+    tagsContainer.appendChild(tagElement);
+    tagList.push(tagElement);
+  });
+}
+
 function tagsToggleFilter(event) {
```
Comment (Adelle-Pitsas) [inline src/scripts.js:191]: I like the split up of the big `tagsToggleFilter` into a smaller function with helper functions. Much more readable!

**Review by grainymac — COMMENTED** [review]
looks good

File: src/scripts.js [hunk 2]
```diff
@@ -142,40 +140,98 @@ function buildTags(recipe) {
...
+  let userTag = getTag(event);
+
+  if (allRecipesTab.checked) {
     const filteredRecipes = recipeRepo.filterByTag(userTag);
-    updateAllRecipeDisplay(filteredRecipes);
+    updateRecipeDisplay(filteredRecipes);
+  }
+  else if (cookbookTab.checked) {
+    const filteredRecipes = user.favoriteRecipeRepo.filterByTag(userTag);
+    updateRecipeDisplay(filteredRecipes);
```
Comment (Adelle-Pitsas) [inline src/scripts.js:216]: Great logic and separation of DOM and data model to display all recipes vs cookbook

File: src/scripts.js [hunk 3]
```diff
@@ -166,6 +166,7 @@ function buildTags(recipe) {
 }
 
 function displayAllTags() {
+  tagsContainer.innerHTML = ''
```
Comment (tricia-holmes) [inline src/scripts.js:169]: Awesome! This fixed the duplicating tags ✅

**Comment by tricia-holmes** [comment]
So cool that the tags are dynamic to the recipe list displayed!

**Pull request merged**